src/OpenRasta/Hosting/HostManager.cs

```diff
@@ -98,7 +98,7 @@ public void Dispose()
     void AssignResolver()
     {
       Resolver = Host.ResolverAccessor?.Resolver ?? new InternalDependencyResolver();
-      Log = Resolver.Resolve<IEnumerable<ILogger>>().LastOrDefault() ?? TraceSourceLogger.Instance;
+      Log = Resolver.Resolve<IEnumerable<ILogger>>().LastOrDefault() ?? DefaultLogger.Instance;

```

Comment:
What will `Resolver.Resolve<IEnumerable<ILogger>>()` resolve by default? An empty collection? A collection containing TraceSourceLogger?

Reply:
1. https://github.com/ololabs/openrasta-core/blob/eeb9f92bfd33c8b088705e5aacf1da3fa48febdc/src/OpenRasta/Configuration/DefaultDependencyRegistrar.cs#L61
2. https://github.com/ololabs/openrasta-core/blob/eeb9f92bfd33c8b088705e5aacf1da3fa48febdc/src/OpenRasta/Configuration/DefaultDependencyRegistrar.cs#L162-L165
3. https://github.com/ololabs/openrasta-core/blob/eeb9f92bfd33c8b088705e5aacf1da3fa48febdc/src/OpenRasta/Configuration/DefaultDependencyRegistrar.cs#L333-L335

AFAIK we don't register an OpenRasta `ILogger`, so it's a `TraceSourceLogger`.

The many different ways `TraceSourceListener` can be resolved is why I added `Level` as a `static` property.

Reply:
For completeness, there's also logic to register known generic loggers (also `TraceSourceLogger<>`).

1. https://github.com/ololabs/openrasta-core/blob/eeb9f92bfd33c8b088705e5aacf1da3fa48febdc/src/OpenRasta/Configuration/DefaultDependencyRegistrar.cs#L78
2. https://github.com/ololabs/openrasta-core/blob/eeb9f92bfd33c8b088705e5aacf1da3fa48febdc/src/OpenRasta/Configuration/DefaultDependencyRegistrar.cs#L240-L246
3. https://github.com/ololabs/openrasta-core/blob/eeb9f92bfd33c8b088705e5aacf1da3fa48febdc/src/OpenRasta/Configuration/DefaultDependencyRegistrar.cs#L325-L331